components/eventing-controller/pkg/backend/jetstreamv2/jetstream_internal_unit_test.go

```diff
@@ -0,0 +1,425 @@
+					subscribeArgs := getSubOptsAsVariadicSlice(jsBackend, sub, eventType)
+
+					jsCtx.On("ConsumerInfo", jsBackend.Config.JSStreamName, jsSubKey.ConsumerName()).Return(nil, nats.ErrConsumerNotFound)
+					jsCtx.On("AddConsumer", jsBackend.Config.JSStreamName, mock.AnythingOfType("*nats.ConsumerConfig")).Return(newConsumer, nil)
+					jsCtx.On("Subscribe", subscribeArgs...).Return(natsSub, nil)
+				}
+			},
+			givenAssertions: func(sub *v1alpha2.Subscription, jsBackend *JetStream, jsCtx *mocks.JetStreamContext, jsSubKey SubscriptionSubjectIdentifier) {
+				// check if the expected methods got called
+				for _, eventType := range sub.Status.Types {
+					jsSubject := jsBackend.getJetStreamSubject(sub.Spec.Source, eventType.CleanType, sub.Spec.TypeMatching)
+					jsSubKey := NewSubscriptionSubjectIdentifier(sub, jsSubject)
+					subscribeArgs := getSubOptsAsVariadicSlice(jsBackend, sub, eventType)
+
+					jsCtx.AssertCalled(t, "ConsumerInfo", jsBackend.Config.JSStreamName, jsSubKey.ConsumerName())
+					jsCtx.AssertCalled(t, "AddConsumer", jsBackend.Config.JSStreamName, mock.AnythingOfType("*nats.ConsumerConfig"))
+					jsCtx.AssertCalled(t, "Subscribe", subscribeArgs...)
+				}
+			},
+		},
+		{
+			name:              "Bind invalid NATS Subscription should be successful",
+			givenSubscription: subWithOneType,
+			givenMocks: func(sub *v1alpha2.Subscription, jsBackend *JetStream, jsCtx *mocks.JetStreamContext, jsSubKey SubscriptionSubjectIdentifier) {
+				for _, eventType := range sub.Status.Types {
+					jsSubject, jsSubKey, newConsumer, natsSub := generateConsumerInfra(jsBackend, sub, eventType)
+
+					// inject invalid subscriber
+					sub := &mocks.Subscriber{}
+					sub.On("IsValid").Return(false)
+					jsBackend.subscriptions[jsSubKey] = sub
+
+					// mock the expected calls
+					jsCtx.On("ConsumerInfo", jsBackend.Config.JSStreamName, jsSubKey.ConsumerName()).Return(newConsumer, nil)
+					jsCtx.On("Subscribe", jsSubject, mock.AnythingOfType("nats.MsgHandler"), mock.AnythingOfType("nats.subOptFn")).Return(natsSub, nil)
+				}
+			},
+			givenAssertions: func(sub *v1alpha2.Subscription, jsBackend *JetStream, jsCtx *mocks.JetStreamContext, jsSubKey SubscriptionSubjectIdentifier) {
+				// check if the expected methods got called
+				for _, eventType := range sub.Status.Types {
+					jsSubject := jsBackend.getJetStreamSubject(sub.Spec.Source, eventType.CleanType, sub.Spec.TypeMatching)
+					jsSubKey := NewSubscriptionSubjectIdentifier(sub, jsSubject)
+
+					jsCtx.AssertCalled(t, "ConsumerInfo", jsBackend.Config.JSStreamName, jsSubKey.ConsumerName())
+					jsCtx.AssertNotCalled(t, "AddConsumer", jsBackend.Config.JSStreamName, mock.AnythingOfType("*nats.ConsumerConfig"))
+					jsCtx.AssertCalled(t, "Subscribe", jsSubject, mock.AnythingOfType("nats.MsgHandler"), mock.AnythingOfType("nats.subOptFn"))
+
+				}
+			},
+		},
+		{
+			name:              "Bind invalid NATS Subscription should result into error",
```

Comment:
I believe we don't need a mock here, but instead a stub. We are testing that a value is returned here, not that an interaction happened. So we can use a stub for both `sub` and `jsCtx`.

Reply:
By not mixing value tests and interaction tests we can also get completely rid of `wantErr` (this is the only value based test which is left in this table test) and the test related code in the table-test body.